app/helpers/rubygems_helper.rb

```diff
@@ -16,11 +16,13 @@ def link_to_page(id, url)
   end
 
   def link_to_github(rubygem)
-    if !rubygem.linkset.code.nil? && URI(rubygem.linkset.code).host == "github.com"
-      URI(@rubygem.linkset.code)
-    elsif !rubygem.linkset.home.nil? && URI(rubygem.linkset.home).host == "github.com"
-      URI(rubygem.linkset.home)
+    results = [rubygem.linkset.code, rubygem.linkset.home].map do |linkset|
+      uri = URI(linkset.to_s)
+
+      uri.host == "github.com" ? uri : nil
     end
+
+    results.reject(&:blank?).first
```

Comment:
Curious about the two steps here? Why assign a variable only to chain off of it anyways?

Reply:
I was grinding gears at 2am trying to get a clean `breaks and return` when we detect our first return in the loop, and gave up in the interest of "figure it out tomorrow". I **know** this is possible, I did it just a few weeks ago in another application, I just honestly can't recall for the life of me The One Weird Trick that makes it work! 😀 

..so the variable is really an artifact of struggling to make it work. I /could/ have chains the pruning of the returned array to the `map` block, but it just felt like I was Doing Too Much (too much "logistical density" about the block, if you follow my meaning). The variable assignment is a waste, but separates the 2 steps cleanly - refine to our possible values, then filter them for what we want.

(Appreciate the feedback, btw - this was really just a code exercise for myself last night, but its prompted some really interesting conversations today!)

Reply:
Ah, yes! That makes sense. I knew I'd get a good answer if I asked.

Were you looking for `Array.find()` by chance?